tests/acl/test_stress_acl.py

```diff
@@ -13,115 +16,162 @@
+    else:
+        dut_port = list(mg_facts['minigraph_portchannels'].keys())[0]
+    if not dut_port:
+        pytest.skip('No portchannels found')
+    if "Ethernet" in dut_port:
+        dut_eth_port = dut_port
+    elif "PortChannel" in dut_port:
+        dut_eth_port = mg_facts["minigraph_portchannels"][dut_port]["members"][0]
+    ptf_src_port = mg_facts["minigraph_ptf_indices"][dut_eth_port]
+
+    topo = tbinfo["topo"]["type"]
+    # Get the list of upstream ports
+    upstream_ports = defaultdict(list)
+    upstream_port_ids = []
+    for interface, neighbor in mg_facts["minigraph_neighbors"].items():
+        port_id = mg_facts["minigraph_ptf_indices"][interface]
+        if (topo == "t1" and "T2" in neighbor["name"]) or (topo == "t0" and "T1" in neighbor["name"]) or \
+                (topo == "m0" and "M1" in neighbor["name"]) or (topo == "mx" and "M0" in neighbor["name"]):
+            upstream_ports[neighbor['namespace']].append(interface)
+            upstream_port_ids.append(port_id)
+
+    return ptf_src_port, upstream_port_ids, dut_port
+
+
+def generate_readd_rules(rand_selected_dut, loops):
+    readd_key = loops + ACL_RULE_NUMS
+    ip_key1 = readd_key / 256
+    ip_key2 = readd_key % 256
+    readd_rule_name = 's/RULE_[0-9]+/RULE_{}/g'.format(readd_key)
+    readd_src_ip = 's/20.0.[0-9]+.[0-9]+/20.0.{}.{}/g'.format(ip_key1, ip_key2)
+    readd_prio = 's/"PRIORITY": "[0-9]+"/"PRIORITY": "{}"/g'.format(readd_key)
+    rand_selected_dut.shell("sed -E -i '{}' {}".format(readd_rule_name, STRESS_ACL_READD_RULE_JSON_FILE))
+    rand_selected_dut.shell("sed -E -i '{}' {}".format(readd_src_ip, STRESS_ACL_READD_RULE_JSON_FILE))
+    rand_selected_dut.shell("sed -E -i '{}' {}".format(readd_prio, STRESS_ACL_READD_RULE_JSON_FILE))
+
+
+def verify_acl_rules(rand_selected_dut, ptfadapter, ptf_src_port,
+                     ptf_dst_ports, acl_rule_list, del_rule_id, verity_status):
+
+    for acl_id in acl_rule_list:
+        ip_key1 = acl_id / 256
+        ip_key2 = acl_id % 256
+
+        src_ip_addr = "20.0.{}.{}".format(ip_key1, ip_key2)
+        dst_ip_addr = "10.0.0.1"
+        pkt = testutils.simple_ip_packet(
+            eth_dst=rand_selected_dut.facts['router_mac'],
+            eth_src=ptfadapter.dataplane.get_mac(0, ptf_src_port),
+            ip_src=src_ip_addr,
+            ip_dst=dst_ip_addr,
+            ip_proto=47,
+            ip_tos=0x84,
+            ip_id=0,
+            ip_ihl=5,
+            ip_ttl=121
+        )
+
+        pkt_copy = pkt.copy()
+        exp_pkt = mask.Mask(pkt_copy)
+        exp_pkt.set_do_not_care_scapy(packet.Ether, 'dst')
+        exp_pkt.set_do_not_care_scapy(packet.Ether, 'src')
+        exp_pkt.set_do_not_care_scapy(packet.IP, "dst")
+        exp_pkt.set_do_not_care_scapy(packet.IP, "ihl")
+        exp_pkt.set_do_not_care_scapy(packet.IP, "tos")
+        exp_pkt.set_do_not_care_scapy(packet.IP, "len")
+        exp_pkt.set_do_not_care_scapy(packet.IP, "id")
```

Comment:
Why do we need to ignore IP.dst, IP.tos, IP.len, IP.ttl and IP.proto?

Reply:
Forgot to delete after test, fixed